### DISCUSSION TITLE
How to use extendPages to add page to a exist router nested route

### DISCUSSION THREAD

**Discussion by Mini-ghost** [nuxt/nuxt#17869] [Questions] [answered]
I need to add a nested page, this page has exactly the same content as another nested page but route and name should different. 

I had tried `extendPages` to add pages but it can't add nested page, like this

```ts
import { defineNuxtModule, extendPages } from '@nuxt/kit';
import { resolve } from 'pathe';

export default defineNuxtModule({
  setup() {
    extendPages(pages => {
      pages.push({
        name: 'angel-nested-page',
        path: '/angel/nested-page',
        file: resolve(__dirname, './../../pages/target/nested-page.vue'),
      });
    });
  },
});
```
`angel` page is already exists, and `angel-nested-page` is nested page

I have no idea to resolve this problem, anyone can  give me some solution or idea?

**Comment by danielroe** [comment, marked as answer]
You can iterate over the array using normal JavaScript to find the route you want to add the nested page to.

**Reply by Mini-ghost** [reply]
Thank you so much, this is very helpful.
